### PR TITLE
#set of minor fixes

### LINKED ISSUE #444: context.py:91: Share base page to refarm side. se2
The puzzle `431-8de31c4b` from #431 has to be resolved: 

https://github.com/fidals/stroyprombeton/blob/23c31673884079ff7f05785f3f67347caf7e8d46/stroyprombeton/context.py#L91-L91

The puzzle was created by duker33 on 19-Feb-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### LINKED ISSUE #447: request_data.py:7: Move Request class to refarm side. se2
The puzzle `431-05aa5c70` from #431 has to be resolved: 

https://github.com/fidals/stroyprombeton/blob/23c31673884079ff7f05785f3f67347caf7e8d46/stroyprombeton/request_data.py#L7-L7

The puzzle was created by duker33 on 19-Feb-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### LINKED ISSUE #450: context.py:107: Create ContextDict type on refarm side. se2
The puzzle `443-35b41d1a` from #443 has to be resolved: 

https://github.com/fidals/stroyprombeton/blob/2dd8517703edfa0b25e9a5a0bdd9dabfdd4d4b4b/stroyprombeton/context.py#L107-L107

The puzzle was created by duker33 on 21-Feb-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### LINKED ISSUE #451: request_data.py:55-57: Check FetchProducts post params....
The puzzle `443-3dae6fc3` from #443 has to be resolved: 

https://github.com/fidals/stroyprombeton/blob/2dd8517703edfa0b25e9a5a0bdd9dabfdd4d4b4b/stroyprombeton/request_data.py#L55-L57

The puzzle was created by duker33 on 21-Feb-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by duker33**
Depends on https://github.com/fidals/refarm-site/pull/307

Closes #444
Closes #450
Closes #451
Closes #447

### CODE REVIEW

File: stroyprombeton/request_data.py [hunk 1]
```diff
@@ -52,10 +40,10 @@ class FetchProducts(Category):
 
     @property
     def id(self):
-        # @todo #443:15m Check FetchProducts post params.
-        #  Raise Http400 if categoryId not exists.
-        #  See `Http400` error on SE for example.
-        return self.request.POST.get('categoryId')
+        id_ = self.request.POST.get('categoryId')
+        if not id_:
+            raise Http400('POST param categoryId is missed')
```
Comment (artemiy312) [inline stroyprombeton/request_data.py]: You should handle this exception in views, otherwise the server will produce 500 response

Comment (duker33) [reply stroyprombeton/request_data.py]: it seems to me that not. So, i'll write test to check this moment

**Comment by duker33** [comment]
depends on https://github.com/fidals/refarm-site/pull/310

File: stroyprombeton/request_data.py (hunk 1 above)
Comment (duker33) [reply stroyprombeton/request_data.py]: seems we have no mech to autocast exceptions to bad request responses.
Fixed code and appended test

File: stroyprombeton/views/catalog.py [hunk 2]
```diff
@@ -11,15 +11,19 @@
...
     """Filter product table on Category page by Name, code, specification."""
-    context_ = stb_context.FetchPositions(
-        request_data.FetchProducts(request, url_kwargs={}),
-    )
+    try:
+        context_ = stb_context.FetchPositions(
+            request_data.FetchProducts(request, url_kwargs={}),
+        )
+    except exception.Http400 as e:
+        return HttpResponseBadRequest(str(e))
```
Comment (artemiy312) [inline stroyprombeton/views/catalog.py]: Let's create a middleware to catch our custom exceptions. It'd be more convenient, robust

**Pull request merged**